src/arch/x86_64/CodeGen.zig

```diff
@@ -4700,7 +4749,9 @@ const CallMCValues = struct {
 };
 
 /// Caller must call `CallMCValues.deinit`.
-fn resolveCallingConventionValues(self: *Self, fn_ty: Type) !CallMCValues {
+fn resolveCallingConventionValues(self: *Self, fn_ty: Type, opts: struct {
+    is_extern_fn: bool,
```

Comment:
You shouldn't need to care whether it's an extern function or not. The frontend should have decided the calling convention by now, and the calling convention is the only information you need to look at to determine where parameters are stored.

Reply:
Oh wait, I could do that by correctly observing C ABI rather than collapsing C ABI with `unspecified`. I obviously need to pass args in registers first when making an extern function call, but I forgot we actually folded the two calling conventions. Thanks for pointing this out!